lib/Target/Btor/BtorToBtorIRTranslation.cpp

```diff
@@ -0,0 +1,781 @@
+                                            opType,
+                                            line->constant[0] - '0'));
+        }
+        break;
+        case BTOR2_TAG_constd: {
+            auto opType = builder.getIntegerType( line->sort.bitvec.width );
+            std::string input( line->constant );
+            res = builder.create<btor::ConstantOp>(unknownLoc, opType, 
+                                        builder.getIntegerAttr(
+                                            opType,
+                                            std::stoi( input )));
+        }
+        break;
+        case BTOR2_TAG_consth: {
+            auto opType = builder.getIntegerType( line->sort.bitvec.width );
+            std::string input( "0x" ); input += line->constant;
+            res = builder.create<btor::ConstantOp>(unknownLoc, opType, 
+                                        builder.getIntegerAttr(
+                                            opType,
+                                            std::stoi( input, nullptr, 16 )));
+        }
+        break;
+
+        case BTOR2_TAG_dec: 
+            res = builder.create<btor::DecOp>(unknownLoc, 
+                                            cache.at(kids[0]));
+        break;
+        case BTOR2_TAG_inc: 
+            res = builder.create<btor::IncOp>(unknownLoc, 
+                                            cache.at(kids[0]));
+        break;
+        case BTOR2_TAG_neg: 
+            res = builder.create<btor::NegOp>(unknownLoc, 
+                                            cache.at(kids[0]));
+        break;
+        case BTOR2_TAG_not: 
+            res = builder.create<btor::NotOp>(unknownLoc, 
+                                            cache.at(kids[0]));
+        break;
+        case BTOR2_TAG_redand: 
+            res = builder.create<btor::RedAndOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]));
+        break;
+        case BTOR2_TAG_redor: 
+            res = builder.create<btor::RedOrOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]));
+        break;
+        case BTOR2_TAG_redxor: 
+            res = builder.create<btor::RedXorOp>(unknownLoc, 
+                                        builder.getIntegerType( 1 ),
+                                        cache.at(kids[0]));
+        break;
+        case BTOR2_TAG_one: {
+            auto opType = builder.getIntegerType( line->sort.bitvec.width );
+            res = builder.create<btor::ConstantOp>(unknownLoc, opType, 
+                                                builder.getIntegerAttr(opType, 1));
+        } 
+        break;
+        case BTOR2_TAG_ones: {
+            auto width = line->sort.bitvec.width;
+            auto opType = builder.getIntegerType( width );
+            auto value = pow(2, width) - 1;
```

Comment:
there should be llvm helper for this, rather than using generic `pow` function.

Reply:
Will take a look